tests/integration-tests/tests/update/test_update_rollback_fixes.py

```diff
@@ -0,0 +1,496 @@
+# This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, express or implied.
+# See the License for the specific language governing permissions and limitations under the License.
+
+"""
+Integration tests for verifying fixes related to cluster update rollback scenarios.
+
+This test validates the following fixes:
+- [F1] clustermgtd remains running after both update and rollback fail
+- [F2] cfn-hup does not enter an endless loop after rollback to a state older than 24h
+- [F3] dna.json files are cleaned up after update failure
+
+Test Plan:
+1. Create cluster with 3 static compute nodes
+2. Inject cfn-signal failure on head node (simulating expired wait condition)
+3. Inject failure on CN1 (disable cfn-hup) BEFORE update
+4. Update cluster config by adding a new queue
+5. Update fails because CN1 won't apply update (cluster readiness check fails)
+6. During update's cluster readiness check (before rollback starts): inject failure on CN2
+   - CN2 has already applied the update at this point
+   - Disabling cfn-hup prevents CN2 from rolling back
+   - Cluster readiness check has 10 attempts (~15 min window)
+7. Rollback starts after update fails
+8. Rollback fails because CN2 won't rollback (still has update target config)
+9. Verify fixes:
+   - clustermgtd is running
+   - dna.json files are deleted
+   - CN3 (healthy node) has correct config version (source config before update)
+   - metadata_db.json is updated
+   - cfn-hup is not in endless loop
+"""
+
+import logging
+import time
+
+import boto3
+import pytest
+from assertpy import assert_that
+from remote_command_executor import RemoteCommandExecutor
+from retrying import retry
+from time_utils import minutes, seconds
+from utils import get_compute_nodes_instance_ids
+
+from tests.common.schedulers_common import SlurmCommands
+
+logger = logging.getLogger(__name__)
+
+
+@pytest.mark.usefixtures("os", "instance", "scheduler")
+def test_update_rollback_fixes(
+    region,
+    pcluster_config_reader,
+    clusters_factory,
+    test_datadir,
+    scheduler_commands_factory,
+):
+    """
+    Test that cluster update rollback fixes work correctly.
+
+    This test verifies:
+    1. clustermgtd remains running after update and rollback failures
+    2. cfn-hup does not enter endless loop when cfn-signal fails
+    3. dna.json files are cleaned up properly
+    """
+    # Create cluster with initial configuration (3 static compute nodes)
+    init_config_file = pcluster_config_reader()
+    cluster = clusters_factory(init_config_file)
+
+    remote_command_executor = RemoteCommandExecutor(cluster)
+    slurm_commands = SlurmCommands(remote_command_executor)
+
+    # Wait for all static nodes to be ready
+    _wait_for_static_nodes_ready(slurm_commands, expected_count=3)
+
+    # Get compute node hostnames
+    compute_nodes = slurm_commands.get_compute_nodes()
+    assert_that(len(compute_nodes)).is_greater_than_or_equal_to(3)
```

Comment:
Why is this assertion required if we already have `_wait_for_static_nodes_ready(slurm_commands, expected_count=3)` above?
If not required, let's remove it